### PR TITLE
feat: recoverable `IllegalWildType` and `UndefinedTypeVar`

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Resolver.scala [hunk 1]
```diff
@@ -2192,9 +2192,11 @@ object Resolver {
   private def semiResolveType(tpe0: NamedAst.Type, wildness: Wildness, env: ListMap[String, Resolution], ns0: Name.NName, root: NamedAst.Root)(implicit level: Level, flix: Flix): Validation[UnkindedType, ResolutionError] = {
     def visit(tpe0: NamedAst.Type): Validation[UnkindedType, ResolutionError] = tpe0 match {
       case NamedAst.Type.Var(ident, loc) =>
-        val symVal = lookupTypeVar(ident, wildness, env)
-        mapN(symVal) {
-          case sym => UnkindedType.Var(sym, loc)
+        lookupTypeVar(ident, wildness, env) match {
+          case Result.Ok(sym) => UnkindedType.Var(sym, loc).toSuccess
+          case Result.Err(e) =>
+            // Note: We assume the default type variable has kind Star.
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:2198]: See other PR; We should have `UnkindedType.Error`

**Review by mlutze — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/language/phase/Resolver.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:2198]: Yeah, but does it solve it or?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:2198]: What I mean is; will we be able to kind it correctly?

Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:2198]: It should eliminate some spurious kind errors

**Pull request merged**